File: Homework/HW2/HW2.py
class HW2:
    def is_bad_version(self, version):
        return False

    def first_bad_version(self, n):
        start, end = 1, n

        while start < end:
            if self.is_bad_version(start):
                return start

            if self.is_bad_version(end - 1):
                end -= 1
            else:
                return end
            start += 1

        return start

File: Homework/HW2/HW2.java
package Homework.HW2;

public class HW2 {
    boolean isBadVersion(int version) {
        return false;
    }

    public int firstBadVersion(int n) {
        int start = 1, end = n;

        while (start < end) {
            if (isBadVersion(start)) {
                return start;
            }

            if (isBadVersion(end - 1)) {
                end--;
            } else {
                return end;
            }
            start++;
        }

        return start;
    }
}
